speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp

```diff
@@ -37,31 +51,31 @@ Recorder::Recorder(const std::string& p_dirName,
     m_recordingDir = std::experimental::filesystem::path(recordingsFolder);
     create_directories(m_recordingDir);
 
-    // Open the files with truncate on, such that if the file was already in use any existing content will be discarded
-    std::experimental::filesystem::path
-        path = std::experimental::filesystem::path(recordingsFolder).append(USER_INPUT_RECORDING_FILE_NAME);
-    std::cout << "Writing user input to " << path << std::endl;
-    m_userInputRecordingFile.open(path, std::ios::binary | std::ios::trunc);
-    m_userInputRecordingFile << std::fixed << std::setprecision(20);
-
-    path = std::experimental::filesystem::path(recordingsFolder).append(DECISIONS_RECORDING_FILE_NAME);
-    std::cout << "Writing decisions to " << path << std::endl;
-    m_decisionsRecordingFile.open(path, std::ios::binary | std::ios::trunc);
-    m_decisionsRecordingFile << std::fixed << std::setprecision(20);
+    CreateRecordingFile(m_recordingDir, m_userInputRecordingFile, USER_INPUT_RECORDING_FILE_NAME);
+    CreateRecordingFile(m_recordingDir, m_decisionsRecordingFile, DECISIONS_RECORDING_FILE_NAME);
+    CreateRecordingFile(m_recordingDir, m_simulationDataRecordingFile, SIMULATION_DATA_RECORDING_FILE_NAME);
 
     // initialize previous input with impossible values. This ensures the first actual values are always written when compression is enabled.
     m_prevUserInput = new float[m_userParamAmount];
     for (int i = 0; i < m_userParamAmount; i++)
     {
         m_prevUserInput[i] = 2.0f;  // 2.0f is impossible user input
     }
```

Comment:
Might be better to use NAN here as well
```suggestion
    {
        m_prevUserInput[i] = NAN;
    }
```